Automattic/woocommerce-payments
# Settings: fix saving settings when enabled payment methods setting contains faulty data

Fixes #4847 

#### Changes proposed in this Pull Request

* When loading the settings page ( `Payments > Settings` ), attempt to repair any potentially corrupted state of the enabled payment method setting.
* Example situation - There were instances of non-US merchants seeing `Error Saving`, since there was `link` appearing as a payment method in their Enabled Payment Methods array

#### Testing instructions

* Enable a US WC Pay account on your test site
* Within `Payment > Settings`, enable `Link by Stripe`, and save settings
* Switch the site to a non-US WC Pay account
* Purge the account cache via WCPay Dev plugin settings to make sure the non-us account takes full effect on the site
* Browse to `Payment > Settings` and click the `Save changes` button at the bottom
* You should no longer see any errors. You should see `Settings Saved.` message. 

-------------------


**Post merge**
